homeassistant/components/madvr/remote.py

```diff
@@ -0,0 +1,87 @@
+"""Support for MadVR remote control."""
+
+from collections.abc import Iterable
+import logging
+from typing import Any
+
+from homeassistant.components.remote import RemoteEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .coordinator import MadVRCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+type MadVRConfigEntry = ConfigEntry[MadVRCoordinator]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: MadVRConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the MadVR remote."""
+    coordinator = entry.runtime_data
+    if not isinstance(coordinator, MadVRCoordinator):
+        raise TypeError("entry.runtime_data is not an instance of MadVRCoordinator")
+
+    async_add_entities(
+        [
+            MadvrRemote(hass, coordinator, entry.entry_id),
+        ]
+    )
+
+
+class MadvrRemote(CoordinatorEntity[MadVRCoordinator], RemoteEntity):
+    """Remote entity for the MadVR integration."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        coordinator: MadVRCoordinator,
+        entry_id: str,
+    ) -> None:
+        """Initialize the remote entity."""
+        super().__init__(coordinator)
+        self.coordinator = coordinator
+        self.madvr_client = self.coordinator.client
+        self._attr_name = self.coordinator.name
+        self._attr_unique_id = f"{entry_id}_remote"
+        self.entry_id = entry_id
+        self._attr_should_poll = False
```

Comment:
Can be set outside of the constructor

Reply:
where?

Reply:
Outside of the constructor as a class variable